src/views/Home/Home.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import track from '../../utils/analytics';
+import store from '../../utils/jwt-store';
+import feature from '../../utils/flaggie';
+import styles from '../../App.module.scss';
+import Search from '../../components/Search';
+import PopularDestinations from '../../components/PopularDestinations';
+import Attractions from '../../components/Attractions';
+import Hero from '../../components/Hero';
+import Modal from '../../components/Modal';
+
+axios.interceptors.request.use((config) => {
+  const newConfig = {
+    ...config,
+    headers: {
+      ...config.headers,
+      env: process.env.REACT_APP_ENV || 'development',
+    },
+  };
+
+  return newConfig;
+});
+
+function Home() {
+  const [state, setState] = useState({
+    clientId: process.env.REACT_APP_OAUTH_GOOGLE_ID,
+    attractions: [],
+    destination: '',
+    isLoading: false,
+    modal: {
+      show: false,
+      attraction: {},
+    },
+    noResults: false,
+  });
+
+  const [features, setFeatures] = useState({
+    loading: true,
+    flags: {},
+  });
+
+  const Feature = feature(features.flags, features.loading);
+
+  // For loading feature flags
+  useEffect(() => {
+    const getFlags = async () => {
+      const promise = new Promise((resolve) => {
+        setTimeout(() => {
+          resolve({
+            'profile-link': false,
+            profile: true,
+          });
+        }, 500);
+      });
+
+      const flags = await promise;
+
+      setFeatures({
+        flags,
+        loading: false,
+      });
+    };
+
+    getFlags();
+  }, []);
+
+  useEffect(() => {
+    const token = store.get();
+    if (token) {
+      setState(prevState => ({
+        ...prevState,
+        loggedIn: true,
+      }));
+    }
+
+    // TODO: This is temporary tracking to validate setup
+    track.pageview('/');
+    track.event({
+      category: 'Main',
+      action: 'Generic Action',
+    });
+  }, []);
+
+  // performs action when user clicks the ROAM button
+  const handleSearch = (destination) => {
+    setState(prevState => ({
+      ...prevState,
+      isLoading: true,
+      destination,
+    }));
+    // request to backend that will request to API and send back the data
+    axios.get(`${process.env.REACT_APP_ENDPOINT}/places/details/${destination}`)
+      .then(({ data: { places } }) => {
+        if (places) {
+          setState(prevState => ({
+            ...prevState,
+            attractions: places,
+            isLoading: false,
+            noResults: false,
+          }));
+        } else {
+          setState(prevState => ({
+            ...prevState,
+            attractions: [],
+            isLoading: false,
+            noResults: true,
+          }));
+        }
+      })
+      .catch((error) => {
+        console.log(error, 'here');  // eslint-disable-line
+      });
+  };
+
+  const showModal = async (place) => {
```

Comment:
Since modal logic will be used in both attractions and profile paths this should exist in App